dragonfly/entity/flammable.go

```diff
@@ -0,0 +1,17 @@
+package entity
+
+// Flammable is an interface for entities that can be set on fire.
+type Flammable interface {
+	// FireProof is whether the entity is currently fire proof.
+	FireProof() bool
+	// FireDamage deals fire damage to the entity.
+	FireDamage(amount float64)
+	// LavaDamage deals lava damage to the entity.
+	LavaDamage(amount float64)
+	// OnFireDuration returns duration of fire in ticks.
+	OnFireDuration() int
+	// SetOnFire sets the entity on fire for the specified duration.
+	SetOnFire(ticks int)
```

Comment:
Ideally these two methods would operate on a `time.Duration`. Ticks are pretty awkward to work with in general for users. You can translate them to ticks in these methods.